src/components/views/DownloadModModal.vue

```diff
@@ -78,63 +78,55 @@ import ProfileModList from '../../r2mm/mods/ProfileModList';
             combo: ThunderstoreCombo,
             store: Store<any>
         ): Promise<void> {
-            return new Promise(async (resolve, reject) => {
+            const assignId = await store.dispatch(
+                'download/addDownload',
+                [`${combo.getMod().getName()} (${combo.getVersion().getVersionNumber().toString()})`]
+            );
 
-                const assignId = await store.dispatch(
-                    'download/addDownload',
-                    [`${combo.getMod().getName()} (${combo.getVersion().getVersionNumber().toString()})`]
+            let downloadedMods: ThunderstoreCombo[] = [];
+            try {
+                downloadedMods = await ThunderstoreDownloaderProvider.instance.download(
+                    profile,
+                    combo,
+                    store.state.download.ignoreCache,
+                    (downloadProgress: number, modName: string, status: number, err: R2Error | null) => {
+                        try {
```

Comment:
Minor: as the error is just re-thrown, using the try-catch block here is just noise.

Reply:
I'll fix this in an upcoming cleanup PR 👍